rpc/common.go

```diff
@@ -133,9 +133,10 @@ func GetGroupControllerCapabilities(ctx context.Context, conn *grpc.ClientConn)
 
 // ProbeForever calls Probe() of a CSI driver and waits until the driver becomes ready.
 // Any error other than timeout is returned.
-func ProbeForever(conn *grpc.ClientConn, singleProbeTimeout time.Duration) error {
+func ProbeForever(ctx context.Context, conn *grpc.ClientConn, singleProbeTimeout time.Duration) error {
```

Comment:
What happens when the context gets canceled? Should `ProbeForever` then return?

That behavior would make sense to me as it would be consistent with other functions accepting a context which then block.

Reply:
Still open.

Reply:
In case of context cancellation, I have implemented the handling as shown below. Would this implementation be appropriate? I would appreciate your guidance if my understanding is incorrect.

```go
select {
	case <-ctx.Done():
		return nil
...
```

Reply:
See https://github.com/kubernetes-csi/csi-lib-utils/pull/149/files#r1584285702